controllers/grid/settings/languages/SubmissionLanguageGridHandler.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * @file controllers/grid/settings/languages/SubmissionLanguageGridHandler.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2000-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionLanguageGridHandler
+ *
+ * @ingroup controllers_grid_settings_languages
+ *
+ * @brief Handle language submission grid requests only.
+ */
+
+namespace PKP\controllers\grid\settings\languages;
+
+use APP\core\Request;
+use APP\core\Services;
+use APP\notification\NotificationManager;
+use PKP\controllers\grid\languages\LanguageGridHandler;
+use PKP\controllers\grid\languages\form\AddLanguageForm;
+use PKP\core\JSONMessage;
+use PKP\core\PKPRequest;
+use PKP\facades\Locale;
+use PKP\linkAction\LinkAction;
+use PKP\linkAction\request\AjaxModal;
+use PKP\notification\PKPNotification;
+use PKP\security\authorization\ContextAccessPolicy;
+use PKP\security\Role;
+
+class SubmissionLanguageGridHandler extends LanguageGridHandler
+{
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->addRoleAssignment(
+            [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN],
+            ['addLanguages', 'addLanguageModal', 'fetchGrid', 'fetchRow']
+        );
+    }
+
+
+    //
+    // Implement methods from GridHandler.
+    //
+    /**
+     * @copydoc GridHandler::authorize()
+     */
+    public function authorize($request, &$args, $roleAssignments)
+    {
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * @copydoc GridHandler::loadData()
+     */
+    protected function loadData($request, $filter)
+    {
+        $context = $request->getContext();
+        $allLocales = Locale::getLocales();
+
+        $addedLocales = $context->getSupportedAddedSubmissionLocales();
```

Comment:
At the beginning getSupportedAddedSubmissionLocales() is null, which needs to be considered here, I think.